### PR TITLE
CELDEV-1094 - extend TabEditor with cel-events for tabedit:successfulSaved

### PR DESCRIPTION
**Pull request by fpichler**
https://synjira.atlassian.net/browse/CELDEV-1094

### CODE REVIEW

File: src/main/webapp/resources/celJS/celTabMenu/TabEditor.js [hunk 1]
```diff
@@ -946,11 +955,11 @@ TE.prototype = {
  saveAndContinue : function(execCallback) {
   const _me = this;
   //TODO add possibility to add JS-listener which can do additional 'isDirty' checks
-  if (this.isDirty()) {
+  if (_me.isDirty()) {
     if(typeof(doBeforeEditSubmit) != 'undefined') {
       doBeforeEditSubmit();
     }
-    var savingDialog = this._getModalDialog();
+    const savingDialog = _me._getModalDialog();
```
Comment (msladek) [inline src/main/webapp/resources/celJS/celTabMenu/TabEditor.js:962]: why is  `_me` needed here? better remove it and replace function expressions within this function with arrow expressions.

File: src/main/webapp/resources/celJS/celTabMenu/TabEditor.js [hunk 2]
```diff
@@ -959,14 +968,20 @@ TE.prototype = {
...
         } else {
+          _me.celFire('tabedit:successfulSaved', { 'jsonResponses' : jsonResponses });
           $('tabMenuPanel').fire('tabedit:successfulSaved', jsonResponses);
         }
+      } catch (exp) {
+        console.error('Saved-listener failed.', exp);
+      }
+      if ((typeof(execCallback) != 'undefined') && execCallback) {
+        execCallback(transport, jsonResponses, failed);
       }
```
Comment (msladek) [inline src/main/webapp/resources/celJS/celTabMenu/TabEditor.js]:
```suggestion
      execCallback && execCallback(transport, jsonResponses, failed);
```

**Review by msladek — CHANGES_REQUESTED** [review]

File: src/main/webapp/resources/celJS/celTabMenu/TabEditor.js (hunk 1 above)
Comment (fpichler) [reply src/main/webapp/resources/celJS/celTabMenu/TabEditor.js:962]: In this method it is needed in line 971, 974 and 977. Hence it is more comprehensible to consequently use it in this method. Consequently I am using it everywhere in a method or nowhere.

Comment (msladek) [reply src/main/webapp/resources/celJS/celTabMenu/TabEditor.js:962]: which wouldn't be needed simply by replacing `function(...)` with `(...) =>` on 969. this change feels like a step backwards.

**Review by msladek — APPROVED** [review]

File: src/main/webapp/resources/celJS/celTabMenu/TabEditor.js (hunk 1 above)
Comment (fpichler) [reply src/main/webapp/resources/celJS/celTabMenu/TabEditor.js:962]: Added CELDEV-1096 to tackle this issue.

**Pull request merged**